src/python_testing/TC_FLABEL_2_1.py

```diff
@@ -0,0 +1,92 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${ALL_CLUSTERS_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import chip.clusters as Clusters
+from chip.interaction_model import Status
+from chip.testing.matter_testing import MatterBaseTest, TestStep, run_if_endpoint_matches, has_attribute, has_cluster, default_matter_test_main
+from mobly import asserts
+
+
+class Test_TC_FLABEL_2_1(MatterBaseTest):
+    def pics_TC_FLABEL_2_1(self) -> list[str]:
+        return ["FLABEL.S"]
+
+    def steps_TC_FLABEL_2_1(self) -> list[TestStep]:
+        return [
+            TestStep(1, "Commission DUT to TH", is_commissioning=True),
+            TestStep(2, "TH reads LabelList from the DUT", "Read is successful"),
+            TestStep(3, "TH tries to write LabelList attribute", "Write fails with UNSUPPORTED_WRITE"),
+            TestStep(4, "Verify LabelList hasn't changed", "LabelList matches initial read")
+        ]
+
+    @run_if_endpoint_matches(has_attribute(Clusters.FixedLabel.Attributes.LabelList) and has_cluster(Clusters.FixedLabel))
```

Comment:
```suggestion
    @run_if_endpoint_matches(has_attribute(Clusters.FixedLabel.Attributes.LabelList))
```
has_attribute will return false if the cluster isn't present, so has_cluster isn't required.